internal/handlers/common/update_array_operators.go

```diff
@@ -91,6 +97,40 @@ func processPopArrayUpdateExpression(doc *types.Document, update *types.Document
 	return changed, nil
 }
 
+// ensureNonExistentPrefix checks if a given path and its prefix
```

Comment:
Based on the description I don't understand what's the difference between this func and `HasByPath`.

Reply:
Added comment in the code :+1: